runtime/lua/vim/lsp/util.lua

```diff
@@ -1720,8 +1779,10 @@ function M.open_floating_preview(contents, syntax, opts)
   local float_option = M.make_floating_popup_options(width, height, opts)
   local floating_winnr = api.nvim_open_win(floating_bufnr, false, float_option)
   if do_stylize then
-    vim.wo[floating_winnr].conceallevel = 2
-    vim.wo[floating_winnr].concealcursor = 'n'
+    vim.wo[floating_winnr].conceallevel =
+      api.nvim_get_option_value('conceallevel', { filetype = 'markdown' })
```

Comment:
Prefer `vim.filetype.get_option` since that caches the values for each filetype.